hash_practice/exercises.py

```diff
@@ -2,18 +2,60 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    dict = {}
+    result = []
+
+    for word in strings:
+        sorted_word = ''.join(sorted(word))
+        if sorted_word in dict:
+                dict[sorted_word].append(word)
+        else:
+            dict[sorted_word] = [word]
+
+    for key, value in dict.items():
+        result.append(value)
+
+    return result
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    d = {}
+    cur_max = 1
+    result = []
+    
+    # create dict to get freqency of each number
+    for n in nums:
+        if n not in d:
+            d[n] = 1
+        else:
+            d[n] += 1
+    
+    while k > 0:

```

Comment:
k never seems to be decremented here for an input of an empty list because if the list is empty so you never get `cur_max = value`.